Sources/Logging/MetadataProvider.swift

```diff
@@ -0,0 +1,118 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Logging API open source project
+//
+// Copyright (c) 2018-2022 Apple Inc. and the Swift Logging API project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift Logging API project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
+import Darwin
+#elseif os(Windows)
+import CRT
+#elseif canImport(Glibc)
+import Glibc
+#elseif canImport(WASILibc)
+import WASILibc
+#else
+#error("Unsupported runtime")
+#endif
+
+#if compiler(>=5.6)
+@preconcurrency public protocol _SwiftLogSendable: Sendable {}
+#else
+public protocol _SwiftLogSendable {}
+#endif
+
+extension Logger {
+    /// A `MetadataProvider` is used to automatically inject runtime-generated metadata
+    /// to all logs emitted by a logger.
+    ///
+    /// ### Example
+    /// A metadata provider may be used to automatically inject metadata such as
+    /// trace IDs:
+    ///
+    /// ```swift
+    /// import Tracing // https://github.com/apple/swift-distributed-tracing
+    ///
+    /// let metadataProvider = MetadataProvider {
+    ///     guard let traceID = Baggage.current?.traceID else { return nil }
+    ///     return ["traceID": "\(traceID)"]
+    /// }
+    /// let logger = Logger(label: "example", metadataProvider: metadataProvider)
+    /// var baggage = Baggage.topLevel
+    /// baggage.traceID = 42
+    /// Baggage.withValue(baggage) {
+    ///     logger.info("hello") // automatically includes ["traceID": "42"] metadata
+    /// }
+    /// ```
+    public struct MetadataProvider: _SwiftLogSendable {
+        /// A default no-op metadata provider, which always returns empty metadata.
+        public static var noop: MetadataProvider {
+            return MetadataProvider { [:] }
+        }
+
+        /// Return the ``Logger/MetadataProvider-swift.struct`` that was configured during ``LoggingSystem/bootstrap(_:)-8ffrb``.
+        public static var bootstrapped: MetadataProvider? {
+            return LoggingSystem.metadataProvider
+        }
+
+        /// Provide ``Logger.Metadata`` from current context.
+        #if swift(>=5.5) && canImport(_Concurrency) // we could instead typealias the function type, but it was requested that we avoid this for better developer experience
+        @usableFromInline
+        internal let _provideMetadata: @Sendable() -> Metadata
+        #else
+        @usableFromInline
+        internal let _provideMetadata: () -> Metadata
+        #endif
+
+        /// Create a new `MetadataProvider`.
+        ///
+        /// - Parameter provideMetadata: A closure extracting metadata from a given `Baggage`.
```

Comment:
What `Baggage`?

Reply:
Oh no, thank you. I'll skim if there's any leftover still